Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketConnection.cs

```diff
@@ -373,10 +375,12 @@ public async Task ValidateCredsAsync()
             token = await AuthTokenManager.GetTokenAsync().ConfigureAwait(false);
         }
 
-        // The token has changed or the connection needs to re-authenticate but can use the same credentials.
-        if (AuthorizationStatus == AuthorizationStatus.AuthorizationExpired || !token.Equals(AuthToken))
+        var authExpired = AuthorizationStatus == AuthorizationStatus.AuthorizationExpired;
+        if (authExpired || !token.Equals(AuthToken))

```

Comment:
Is there a situation here where the code would need to handle a null token being supplied from the AuthTokenManager. Previous code threw an exception in that case (see line 354 for conditional on null check).

Reply:
Added a null check.